File: LeetCode_Solutions/LeetCode_Solutions/ClimbingStairs_70.m
//
//  ClimbingStairs_70.m
//  LeetCode_Solutions
//
//  Created by 来自遥远星系的核心巡洋舰 on 16/11/11.
//  Copyright © 2016年 KnightJoker. All rights reserved.
//




#import "ClimbingStairs_70.h"

@implementation ClimbingStairs_70

/*  递归解法
 
 - (NSInteger)climbStairs:(NSInteger)n {
 
 if (n < 2 || n == 2) {
    return n;
 }
    return [self climbStairs:(n-1)] + [self climbStairs:(n-2)];
 }
 
 */


- (NSInteger)climbStairs:(NSInteger)n {

    NSInteger stepone = 0;
    NSInteger steptwo = 1;
    NSInteger sum = 0;
    
    for(NSInteger i = 0;i < n;i++) {
        sum = stepone + steptwo;
        stepone = steptwo;
        steptwo = sum;
        
    }
    
    return sum;

}

@end

File: LeetCode_Swift/LeetCode_Swift/ClimbingStairs_70.swift
//
//  ClimbingStairs_70.swift
//  LeetCode_Swift
//
//  Created by Huni on 22/08/2017.
//  Copyright © 2017 KnighhtJoker. All rights reserved.
//

import Foundation

public class ClimbingStairs_70 {
    func climbStairs(_ n: Int) -> Int {
        var stepOne = 0
        var stepTwo = 1
        var sum = 0
        var i = 0
        while i < n {
            sum = stepOne + stepTwo
            stepOne = stepTwo
            stepTwo = sum
            i += 1
        }
        
        return sum
    }
}
